meetings/2022-11/dec-01.md

```diff
@@ -0,0 +1,1104 @@
+
+MF: Yeah, thank you for the feedback. I was somewhat on the fence myself. I chose to lean toward the consistency between flatMap and from, because I think that might be what developers value. I feel it is a fairly rare case but I agree, it’s a surprising case where Iterator.from(string) throws. I think I would be open to changing that if we have others who agree and not many who disagree. I would assume you also want AsyncIterator.from to support strings?
+
+KG: Well, yes, exactly the same comment.
+
+MF: I'd be open.
+
+JHD: Yeah, I agree with KG. I think that it's bizarre that strings are iterable but either way you're not from string. It makes a lot of sense.
+
+BT:  All right. Thank you, Justin.  \
+ \
+JRL: Weakly kind of like the current behavior. It just makes it really easy to explain how flatMap flattening behavior will work. Whatever `from` gives you, if it turns into a real iterator or returns multiple results, that's how flatten would work. If it only returns a single item, then that’s how flatten will work, they'll just be a single item. It kind of makes it nice and easy to explain. I'm just curious how we explain if we go with this change, how do you want to explain flatten's behavior.
+
+MF: It doesn't make it simpler, it just stays the same. It's just that the description of Iterator.from would be simpler if you were to defer to that. So if you were to instead define flatMap flattening behavior in terms of Iterator.from, just have the exception in flatMap instead of from. I don't think it changes the complexity of describing that, just where you describe that.
+
+YSV: On the topic of Function.prototype versus AsyncFunction.prototype. I agree with the arguments that Michael made that we should be going with function prototype and I support that. okay?
+
+MF: That was very quiet. It sounded like support for Function.prototype for built-in async functions.
+
+BT: Shu. agreeing with the statement. It sounds like
+
+SYG: Yeah, that's it.
+
+BT: Thank you, Shu.
+
+NRO: We already have different functions like the promise combinator. So you like faction web platform the old have functions process like I would find it, strange to her different prototype, just because it deserves, like, newer methods are assumed from this.
+
+BT: Okay. thank you. JHD. is also plus one on history,
+
+JHD: I just want to add that I think if async functions had been around in es6, we would have made all promise returning functions have AsyncFunction.prototype. But since we didn't, we kept with that consistency when we added a new promise combinator's and things like that. and it makes sense to keep with that consistency even though it's like I would say it's argued a little weird that built-in promise-returning functions are not AsyncFunctions, but it's not that important and we should shouldn't start doing something something different.
```

Comment:
```suggestion
JHD: I just want to add that I think if async functions had been around in es6, we would have made all promise returning functions have AsyncFunction.prototype. But since we didn't, we kept with that consistency when we added a new promise combinators and things like that. It makes sense to keep with that consistency even though I would argue it's a little weird that built-in promise-returning functions are not AsyncFunctions, but it's not that important, and we should shouldn't start doing something different.
```